kore/src/Kore/JsonRpc.hs

```diff
@@ -143,7 +141,10 @@ data HaltReason
 
 data ExecuteResult = ExecuteResult
     { reason :: HaltReason
-    , states :: [ExecuteState]
+    , depth :: !Depth
+    , state :: ExecuteState
+    , nextStates :: Maybe [ExecuteState]
```

Comment:
Is there a difference in meaning when `nextState` is `Nothing` vs `Just []` or is this purely so that the `ToJSON` instance does not include the `nextState` when `Nothing`?

Reply:
Just for JSON. We can of course make it non-optional but it is clearly only needed for some of the reasons.

Reply:
No I think that's absolutely fine. Just wanted to check. Thanks